compiler/native/compile.go

```diff
@@ -5,14 +5,86 @@
 package native
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"net/http"
 	"strings"
 
+	"github.com/go-vela/types/library"
 	"github.com/go-vela/types/pipeline"
+	"github.com/go-vela/types/yaml"
 )
 
+// ModifyRequest contains the payload passed to the modification endpoint
+type ModifyRequest struct {
+	Pipeline *yaml.Build `json:"pipeline,omitempty"`
+	Build    int         `json:"build,omitempty"`
+	Repo     string      `json:"repo,omitempty"`
+	User     string      `json:"user,omitempty"`
+}
+
+// modifyConfig sends the configuration to external http endpoint for modification.
+func (c *client) modifyConfig(build *yaml.Build, libraryBuild *library.Build, repo *library.Repo) (*yaml.Build, error) {
+	// create request to send to endpoint
+	modReq := &ModifyRequest{
+		Pipeline: build,
+		Build:    libraryBuild.GetNumber(),
+		Repo:     repo.GetName(),
+		User:     libraryBuild.GetAuthor(),
+	}
+
+	// marshal json to send in request
+	b, err := json.Marshal(modReq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal modify payload")
+	}
+
+	// create POST request
+	req, err := http.NewRequest("POST", c.ModificationService.Endpoint, bytes.NewBuffer(b))
+	if err != nil {
+		return nil, err
+	}
+
+	// create the client with timeouts
+	client := &http.Client{
+		Timeout: c.ModificationService.Timeout,
+	}
+
+	// add content-type and auth headers
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.ModificationService.Secret))
+
+	// send the request
+	resp, err := client.Do(req)
```

Comment:
Would you ever be interested in making these calls with a backoff policy? It's a question we've been thinking about for functionality that has to reach out to an external service

Reply:
Are you thinking about using duplicating something like https://github.com/go-vela/server/blob/5d651bc423fd5344bcff923127c07e765a755ded/source/github/repo.go#L21?

Reply:
We have it here as well:
https://github.com/go-vela/pkg-runtime/blob/master/runtime/kubernetes/container.go#L300-L310

I think we'd probably wanna think about picking a library to standardize on usage but I more just meant do you see a need for it with these types of calls?

Reply:
I agree that some backoff policy/logic should be added to help reduce the failure scenarios. My only concern is the webhook timeouts I mentioned on above.

Reply:
Agreed, we'd have to set hard limits. Should Vela check the timeout limits when an admin sets them? That way we stay within GitHub's timeouts and the server doesn't even start with bad configuration.

Reply:
> Should Vela check the timeout limits when an admin sets them?

I think that makes sense. Any thoughts if we should distinguish between the github.com and GitHub Enterprise webhook timeouts? I'm initially thinking we conform to the github.com timeout of 10 seconds.

Reply:
I'm good with the 10 second time validation

Reply:
Have you looked into a library for the backoff policy yet? I've had decent success with https://github.com/hashicorp/go-retryablehttp for standard http calls.